src/v/resource_mgmt/memory_groups.cc

```diff
@@ -43,11 +43,30 @@ struct memory_shares {
     }
 };
 
+size_t compaction_reservation() {
+    const auto& cluster_cfg = config::shard_local_cfg();
+    if (!cluster_cfg.log_compaction_use_sliding_window.value()) {
+        return 0;
+    }
+    auto pct_limit_opt
+      = cluster_cfg.storage_compaction_key_map_memory_limit_percent.value();
+    auto bytes_cfg = cluster_cfg.storage_compaction_key_map_memory.value();
+    if (!pct_limit_opt.has_value()) {
+        return bytes_cfg;
+    }
+    uint64_t bytes_limit = ss::memory::stats().total_memory()
+                           * pct_limit_opt.value();
```

Comment:
Would we want to account for WebAssembly before this?

Reply:
Good question. I'd originally done it this way to be easier for users to reason about what the config actually means. OTOH accounting for WASM would be a more definitive safeguard against misconfiguration.

I don't think users should have to think about this config, and should mostly rely on the `key_map_memory`, and this is only important in memory-constrained environments where it's better to err on the side of conservatism. Will update.